app/upgrades/v16/upgrades.go

```diff
@@ -26,6 +27,11 @@ func CreateUpgradeHandler(
 		// Enable ICA controller
 		keepers.ICAControllerKeeper.SetParams(ctx, icacontrollertypes.DefaultParams())
 
+		// Set default blocks per epoch
+		providerParams := keepers.ProviderKeeper.GetParams(ctx)
+		providerParams.BlocksPerEpoch = providertypes.DefaultBlocksPerEpoch
+		keepers.ProviderKeeper.SetParams(ctx, providerParams)
+
```

Comment:
The simplest way to do it:
```
subspace := keepers.GetSubspace(providertypes.ModuleName)
subspace.Set(ctx, providertypes.KeyBlocksPerEpoch, int64(providertypes.DefaultBlocksPerEpoch))
```

Reply:
The upgrade code is in https://github.com/cosmos/interchain-security/pull/1757

It will should be available in the next release candidate of `ICS@v4.1.x`

Reply:
Okay, that makes sense to do the upgrade in ICS. So why is the `BlocksPerEpoch` param set to default here (again)?

Reply:
Making it explicit. Otherwise you have to reference ICS docs - reviewind and tracking changelogs becomes a pain.

We did a similar thing for ICA controller.